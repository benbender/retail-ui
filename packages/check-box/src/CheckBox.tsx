import { useThemeCtx } from '@retail-ui/theme'
import clsx from 'clsx'
import * as React from 'react'

type Ref = HTMLInputElement
type ReactInputProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'checked'
>

export type CheckBoxProps = {
  isChecked?: boolean
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
}

export const CheckBox = React.forwardRef<Ref, ReactInputProps & CheckBoxProps>(
  (props, ref) => {
    const { children, className, isChecked = false, onChange, ...rest } = props
    const {
      theme: { CheckBoxStyles },
    } = useThemeCtx()

    const cls = clsx(className, CheckBoxStyles.base)

    return (
      <input
        ref={ref}
        type="checkbox"
        className={cls}
        checked={isChecked}
        onChange={onChange}
        {...rest}
      />
    )
  },
)

CheckBox.displayName = 'CheckBox'
