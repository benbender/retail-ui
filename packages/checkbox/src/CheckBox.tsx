import clsx from 'clsx'
import * as React from 'react'

import { CheckBoxStyles } from './styles'

type Ref = HTMLInputElement
type ReactInputProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'disabled'
>

export type CheckBoxProps = {
  isDisabled?: boolean
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
  label?: React.ReactNode
}

export const CheckBox = React.forwardRef<Ref, ReactInputProps & CheckBoxProps>(
  (props, ref) => {
    const { className, isDisabled, onChange, label, ...rest } = props

    const cls = clsx(
      className,
      CheckBoxStyles.base,
      isDisabled && CheckBoxStyles.disabled,
    )

    const labelCls = clsx(CheckBoxStyles.label, CheckBoxStyles)

    return (
      <label className={labelCls}>
        <input
          ref={ref}
          type="checkbox"
          className={cls}
          onChange={onChange}
          {...rest}
        />
        {label && (
          <span
            className={`ml-2 ${isDisabled && `opacity-50 cursor-not-allowed`}`}
          >
            {label}
          </span>
        )}
      </label>
    )
  },
)

CheckBox.displayName = 'CheckBox'
