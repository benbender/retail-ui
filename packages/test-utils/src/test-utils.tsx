import { defaultTheme, ThemeProvider } from '@retail-ui/theme'
import {
  render as RtlRender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react'
import * as React from 'react'

type UI = Parameters<typeof RtlRender>[0]

// UI-less passthrough fallback to prevent using conditional logic in render
function ChildrenPassthrough({ children }: { children: React.ReactElement }) {
  return children
}

export interface TestOptions extends Omit<RenderOptions, 'wrapper'> {
  /**
   * optional additional wrapper, e.g. Context
   *
   * @example
   * ```ts
   * // single wrapper
   * render(<MyConponent />, {
   *  wrapper: MyContext
   * });
   *
   * // multiple wrapper
   * render(<MyConponent />, {
   *  wrapper: ({ children }) => (
   *    <ContextA>
   *      <ContextB>
   *        {children}
   *      <ContextB />
   *    <ContextA />
   *  )
   * });
   *
   * ```
   */
  wrapper?: typeof ChildrenPassthrough
}

/**
 * Custom render for @testing-library/react
 *
 * @see https://testing-library.com/docs/react-testing-library/setup#custom-render
 * @param component the component under test
 * @param options customized test options
 */
export const render = (
  ui: UI,
  { wrapper: Wrapper = ChildrenPassthrough, ...options }: TestOptions = {},
): RenderResult => {
  return RtlRender(
    <ThemeProvider value={{ theme: defaultTheme }}>
      <Wrapper>{ui}</Wrapper>
    </ThemeProvider>,
    options,
  )
}

export { fireEvent } from '@testing-library/react'
