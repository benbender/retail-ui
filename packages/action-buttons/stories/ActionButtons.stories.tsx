import * as React from 'react'

import { ActionButtons, ActionButtonsProps } from '../src'

const meta = {
  title: 'Retail-UI/ActionButtons',
  component: ActionButtons,
}

export default meta

const Template = (args: ActionButtonsProps) => (
  <ActionButtons {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {}
