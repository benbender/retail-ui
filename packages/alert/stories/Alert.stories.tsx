import * as React from 'react'

import { Alert, AlertProps } from '../src'

const meta = {
  title: 'Retail-UI/Alert',
  component: Alert,
}

export default meta

const Template = (args: AlertProps) => (
  <Alert {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  children: 'A alert that informs you of stuff',
}
