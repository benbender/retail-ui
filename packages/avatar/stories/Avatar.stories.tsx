import * as React from 'react'

import { Avatar, AvatarProps } from '../src'

const meta = {
  title: 'Retail-UI/Avatar',
  component: Avatar,
}

export default meta

const Template = (args: AvatarProps) => (
  <Avatar {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  src: `https://pbs.twimg.com/profile_images/1285102166709919744/XvlWrJny_400x400.jpg`,
}
