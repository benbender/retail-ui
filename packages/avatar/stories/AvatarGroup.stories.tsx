import * as React from 'react'

import { Avatar, AvatarGroup, AvatarProps } from '../src'

const meta = {
  title: 'Retail-UI/AvatarGroup',
  component: AvatarGroup,
}

export default meta

const src = `https://pbs.twimg.com/profile_images/1285102166709919744/XvlWrJny_400x400.jpg`

const Group = (args: AvatarProps) => (
  <AvatarGroup data-testid={meta.title}>
    <Avatar src={src} {...args} />
    <Avatar title="PJ" size="base" {...args} />
    {/* <Avatar icon={UserSolid} {...args} /> */}
  </AvatarGroup>
)

export const Simple = Group.bind({})
// @ts-ignore
Simple.args = {}
