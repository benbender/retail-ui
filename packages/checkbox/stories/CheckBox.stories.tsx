import { Meta, Story } from '@storybook/react/types-6-0'
import * as React from 'react'

import { CheckBox, CheckBoxProps } from '../src'

const meta: Meta = {
  title: 'Retail-UI/CheckBox',
  component: CheckBox,
}

export default meta

const Template: Story<CheckBoxProps> = (args) => (
  <CheckBox
    label="CheckBox"
    data-testid={meta.title}
    {...args}
    onChange={() => {}}
  />
)

export const Simple = Template.bind({})

Simple.args = {}
