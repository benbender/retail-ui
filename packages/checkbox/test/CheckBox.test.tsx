import { render } from '@retail-ui/test-utils'
import * as React from 'react'

import * as CheckBox from '../stories/CheckBox.stories'

describe('CheckBox', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<CheckBox.Simple />)
    expect(asFragment()).toMatchSnapshot()
  })
})
