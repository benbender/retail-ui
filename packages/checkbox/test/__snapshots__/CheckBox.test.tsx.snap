// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders correctly 1`] = `
<DocumentFragment>
  <label
    class="flex items-center text-gray-600 dark:text-gray-400 base disabled label"
  >
    <input
      class="text-purple-600 form-checkbox focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray dark:border-gray-600 dark:bg-gray-700 dark:focus:border-gray-600"
      data-testid="Retail-UI/CheckBox"
      type="checkbox"
    />
    <span
      class="ml-2 undefined"
    >
      CheckBox
    </span>
  </label>
</DocumentFragment>
`;
